variables-common.tf

```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Copyright (c) 2022, 2023 Oracle Corporation and/or its affiliates.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
 
 locals {
```

Comment:
Variables files should contain only variable definitions. Please move locals into a locals.tf

Reply:
Also consider moving all locals here into 1 file and document the subsections. Otherwise, it's tedious to hunt for a local variable

Reply:
It can be opinionated, and I think there are good reasons to colocate sections when they are closely related. There are always exceptions, but this also isn't visible outside of the module and doesn't impact usage or presentation for users one way or another. I do agree that variables should be isolated though, and only use locals in this context when strictly derived from one/more in the same file to be used in place of them, as is the case with this fallback logic here - the vars shouldn't be used directly outside of the file.
In many (I'd argue most) cases, a pattern of organizing strictly on block type ends up even more disorganized with [tight coupling/low cohesion](https://en.wikipedia.org/wiki/Coupling_(computer_programming)) and some pretty long files to scroll through, e.g.  in `4.x` [modules/extensions/templates.tf](https://raw.githubusercontent.com/oracle-terraform-modules/terraform-oci-oke/c67a0a17a25f42c3793e3c5ee2e0aef8fe3c70fd/modules/extensions/templates.tf), [modules/extensions/locals.tf](https://raw.githubusercontent.com/oracle-terraform-modules/terraform-oci-oke/c67a0a17a25f42c3793e3c5ee2e0aef8fe3c70fd/modules/extensions/locals.tf), [modules/network/locals.tf](https://raw.githubusercontent.com/oracle-terraform-modules/terraform-oci-oke/c67a0a17a25f42c3793e3c5ee2e0aef8fe3c70fd/modules/network/locals.tf), [modules/network/nsgs.tf](https://raw.githubusercontent.com/oracle-terraform-modules/terraform-oci-oke/c67a0a17a25f42c3793e3c5ee2e0aef8fe3c70fd/modules/network/nsgs.tf), etc.
If a file can cleanly encapsulate local values sourced from/used by resources nearby and their outputs, it cuts down on needing to hop around looking for them when making a change and on the number of files affected. No solution is perfect, and there's always Find in Files/grep :shrug:

<img src="https://upload.wikimedia.org/wikipedia/commons/thumb/0/09/CouplingVsCohesion.svg/1920px-CouplingVsCohesion.svg.png" width=50% height=50%>

Reply:
Discussed and in agreement.